<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ContrastDialog</class>
 <widget class="QDialog" name="ContrastDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>929</width>
    <height>476</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background:rgb(53, 53, 53)
</string>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>540</x>
     <y>430</y>
     <width>341</width>
     <height>32</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
   </property>
  </widget>
  <widget class="QLabel" name="histogramLabel">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>621</width>
     <height>211</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>240</y>
     <width>277</width>
     <height>221</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QRadioButton" name="linearRadioButton">
      <property name="styleSheet">
       <string notr="true">color:white
</string>
      </property>
      <property name="text">
       <string>Linear Transformation</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QRadioButton" name="piecewiseRadioButton">
      <property name="styleSheet">
       <string notr="true">color:white</string>
      </property>
      <property name="text">
       <string>Piecewise Linear Transformation</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSlider" name="faSlider">
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSlider" name="fbSlider">
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSlider" name="gaSlider">
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSlider" name="gbSlider">
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSlider" name="gminSlider">
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSlider" name="gmaxSlider">
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="functionLabel">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>250</y>
     <width>291</width>
     <height>201</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="equalizationPushButton">
   <property name="geometry">
    <rect>
     <x>690</x>
     <y>380</y>
     <width>191</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Histogram Equalization</string>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>30</y>
     <width>266</width>
     <height>311</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QRadioButton" name="logarithmRadioButton">
      <property name="styleSheet">
       <string notr="true">color:white
</string>
      </property>
      <property name="text">
       <string>Logarithm Transformation</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label">
      <property name="styleSheet">
       <string notr="true">color:white</string>
      </property>
      <property name="text">
       <string>s = a + clog( b + r )</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
        <property name="text">
         <string> a</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QSlider" name="logaSlider">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QSlider" name="logcSlider">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QSlider" name="logbSlider">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="logbSpinBox">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="logcSpinBox">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_4">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
        <property name="text">
         <string> b </string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_5">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
        <property name="text">
         <string> c</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="logaSpinBox">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QRadioButton" name="expotentRadioButton">
      <property name="styleSheet">
       <string notr="true">color:white</string>
      </property>
      <property name="text">
       <string>Expotent Transformation</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_2">
      <property name="styleSheet">
       <string notr="true">color:white</string>
      </property>
      <property name="text">
       <string>s = c( a + r )^ b</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="1" column="0">
       <widget class="QLabel" name="label_6">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
        <property name="text">
         <string> a</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QSlider" name="expaSlider">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QSlider" name="expcSlider">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QSlider" name="expbSlider">
        <property name="minimum">
         <number>50</number>
        </property>
        <property name="maximum">
         <number>150</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="expcSpinBox">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_7">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
        <property name="text">
         <string> b </string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_8">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
        <property name="text">
         <string> c</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="expaSpinBox">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="expbSpinBox">
        <property name="styleSheet">
         <string notr="true">color:white</string>
        </property>
        <property name="minimum">
         <double>0.500000000000000</double>
        </property>
        <property name="maximum">
         <double>1.500000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ContrastDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ContrastDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
